Did not reinstall my global modules

This script is great, thanks! But there should perhaps be a warning about reinstalling global modules. My global modules were somewhere in /usr/local (maybe /usr/local/lib; I think it came bundled with a node version installed by brew cask) and they didn't get reinstalled. Prior to running this script, my npm had been broken to the point where no npm executable was in the path after running `npm update -g`, and I didn't dig around enough to create a new symlink. I backed them up to a file before running the script so I know what I'm missing.

Bump for interest!

```
~ ❯❯❯ node-reinstall 0.12.4                                                                                                                                    ⏎
Completely reinstalling Node, npm.
=> Downloading nvm from git to '/Users/flo/.nvm'
=> Cloning into '/Users/flo/.nvm'...
remote: Counting objects: 3693, done.
remote: Total 3693 (delta 0), reused 0 (delta 0), pack-reused 3693
Receiving objects: 100% (3693/3693), 827.00 KiB | 169.00 KiB/s, done.
Resolving deltas: 100% (2132/2132), done.
Checking connectivity... done.
* (detached from v0.25.1)
  master

=> Source string already in /Users/flo/.bashrc
=> Close and reopen your terminal to start using nvm
######################################################################## 100.0%
Now using node v0.12.4 (npm v2.10.1)
default -> 0.12.4 (-> v0.12.4)
Reinstalling your global npm modules:

npm ERR! addLocal Could not install /Users/flo
npm ERR! Darwin 14.3.0
npm ERR! argv "/Users/flo/.nvm/versions/node/v0.12.4/bin/node" "/Users/flo/.nvm/versions/node/v0.12.4/bin/npm" "install" "--global"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1
npm ERR! code EISDIR
npm ERR! errno -21

npm ERR! EISDIR, read
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/flo/npm-debug.log
node-reinstall is done. 👍
~ ❯❯❯ less /Users/flo/npm-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/Users/flo/.nvm/versions/node/v0.12.4/bin/node',
1 verbose cli   '/Users/flo/.nvm/versions/node/v0.12.4/bin/npm',
1 verbose cli   'install',
1 verbose cli   '--global' ]
2 info using npm@2.10.1
3 info using node@v0.12.4
4 verbose install initial load of /Users/flo/.nvm/versions/node/v0.12.4/lib/package.json
5 verbose readDependencies loading dependencies from /Users/flo/.nvm/versions/node/v0.12.4/lib/package.json
6 silly cache add args [ '.', null ]
7 verbose cache add spec .
8 silly cache add parsed spec { raw: '.',
8 silly cache add   scope: null,
8 silly cache add   name: null,
8 silly cache add   rawSpec: '.',
8 silly cache add   spec: '/Users/flo',
8 silly cache add   type: 'local' }
9 error addLocal Could not install /Users/flo
10 verbose stack Error: EISDIR, read
10 verbose stack     at Error (native)
11 verbose cwd /Users/flo
12 error Darwin 14.3.0
13 error argv "/Users/flo/.nvm/versions/node/v0.12.4/bin/node" "/Users/flo/.nvm/versions/node/v0.12.4/bin/npm" "install" "--global"
14 error node v0.12.4
15 error npm  v2.10.1
16 error code EISDIR
17 error errno -21
18 error EISDIR, read
19 error If you need help, you may report this error at:
19 error     <https://github.com/npm/npm/issues>
20 verbose exit [ -21, true ]
```

fyi - I think the reinstall caused some troubles in my config.

```
npm ERR! Attempt to unlock ~/myapp/node_modules/jasmine-expect, which hasn't been locked
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>
npm WARN locking Error: EACCES, open '/Users/flo/.npm/_locks/superagent-20b9725d17d0bfba.lock'
npm WARN locking     at Error (native)
npm WARN locking  /Users/flo/.npm/_locks/superagent-20b9725d17d0bfba.lock failed { [Error: EACCES, open '/Users/flo/.npm/_locks/superagent-20b9725d17d0bfba.lock']
npm WARN locking   errno: -13,
npm WARN locking   code: 'EACCES',
npm WARN locking   path: '/Users/flo/.npm/_locks/superagent-20b9725d17d0bfba.lock' }
npm ERR! Darwin 14.3.0
npm ERR! argv "/Users/flo/.nvm/versions/node/v0.12.4/bin/node" "/Users/flo/.nvm/versions/node/v0.12.4/bin/npm" "install"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1

npm ERR! Attempt to unlock ~/myapp/node_modules/superagent, which hasn't been locked
npm ERR!
npm ERR! If you need help, you may report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! Please include the following file with any support request:
npm ERR!     ~/myapp/npm-debug.log
```

Found a post on Stack Overflow that solved it with:
~/myapp ❯❯❯ sudo chown -R $USER ~/.npm                                         ⏎ ✱ ◼
Password:
~/myapp ❯❯❯ sudo chown -R $USER /usr/local/lib/node_modules                      ✱ ◼
chown: /usr/local/lib/node_modules: No such file or directory
~/myapp ❯❯❯ sudo chown -R $USER /usr/local
```

--- a/README.md
+++ b/README.md
@@ -29,9 +29,11 @@ Whenever you feel like you need to completely re-install Node and NPM, simply ru
 ## Commands
 
 	node-reinstall					re-install node and npm using nvm
-	node-reinstall -h [--help]		show help
-	node-reinstall -v [--version]	show the node-reinstall version number
+	node-reinstall [-h|--help]		show help
+	node-reinstall [-v|--version]	show the node-reinstall version number
+	node-reinstall [-f|--force]		installs defaults without user confirmation
 	node-reinstall --nave			re-install using nave
 	node-reinstall --nvm			re-install using stable nvm - the default
 	node-reinstall --nvm-latest		re-install using latest nvm - creationix/nvm:master
 	node-reinstall 0.12				specify a default node version - currently 0.10
+

--- a/node-reinstall.sh
+++ b/node-reinstall.sh
@@ -3,7 +3,7 @@
 # credit: http://stackoverflow.com/a/11178106/2083544
 
 ## program version
-VERSION="0.0.8"
+VERSION="0.0.9"
 
 ## path prefix
 PREFIX="${PREFIX:-/usr/local}"
@@ -11,6 +11,9 @@ PREFIX="${PREFIX:-/usr/local}"
 ## version control systems
 USE_NAVE=0
 USE_NVM=1
+
+# use --force to bypass user confirmation
+FORCE=0
 
 ## default node version
 NODE_VERSION="0.10"
@@ -25,8 +28,9 @@ usage () {
   printf "%s\n" "Commands:"
   printf "\n"
   printf "\t%s\t\t\t%s\n" "node-reinstall" "re-install node and npm using nvm"
-  printf "\t%s %s %s\t%s\n" "node-reinstall" "-h" "[--help]" "show help"
-  printf "\t%s %s %s\t%s\n" "node-reinstall" "-v" "[--version]" "show the node-reinstall version number"
+  printf "\t%s %s\t%s\n" "node-reinstall" "[-h|--help]" "show help"
+  printf "\t%s %s\t%s\n" "node-reinstall" "[-v|--version]" "show the node-reinstall version number"
+  printf "\t%s %s\t%s\n" "node-reinstall" "[-f|--force]" "installs defaults without user confirmation"
   printf "\t%s %s\t\t%s\n" "node-reinstall" "--nave" "re-install using nave"
   printf "\t%s %s\t\t%s\n" "node-reinstall" "--nvm" "re-install using stable nvm - the default"
   printf "\t%s %s\t%s\n" "node-reinstall" "--nvm-latest" "re-install using latest nvm - creationix/nvm:master"
@@ -47,6 +51,10 @@ usage () {
       --version|-v)
         echo ${VERSION}
         exit
+        ;;
+
+      --force|-f)
+        FORCE=1
         ;;
 
       --nave)
@@ -93,6 +101,36 @@ else
   if [[ -n $GLOBAL_MODULES ]]; then
     echo "Will reinstall these global npm modules:"
     echo $GLOBAL_MODULES
+  else
+    echo "===== ALERT! ====="
+    echo "The script did not find any global node modules (npm -g list)"
+    echo "If you are sure you installed global node modules"
+    echo "(by running npm install -g some-module), you might want to stop "
+    echo "here and locate those, because they won't be re-installed,"
+    echo "and since we'll be deleting all the possible install paths "
+    echo "that most people could use, you probably won't find them."
+    echo ""
+    echo "This can sometimes happen if you've installed global node modules"
+    echo "under a different node environment (for example, using nvm or nave)."
+    echo "It might help to run: "
+    echo "history | grep 'npm install' and look for either -g or --global"
+    echo ""
+    echo "If you aren't really sure, or you are sure and don't care, "
+    echo "you can continue; we'll re-install things the proper way, and"
+    echo "the next time you run this script you'll see a list of "
+    echo "any global node modules you've installed since now."
+  fi
+
+  if [[ $FORCE == 0 ]]; then
+    echo ""
+    echo "Would you like to continue running node-reinstall?"
+    echo ""
+    select yn in "Yes" "No"; do
+          case $yn in
+              Yes ) break;;
+              No ) exit;;
+          esac
+    done
   fi
 fi
 
@@ -138,10 +176,21 @@ elif (( $USE_NAVE )); then
   nave usemain $NODE_VERSION
 fi
 
-if [ ${#GLOBAL_MODULES[@]} -gt 0 ]; then
+if [[ ! -z ${GLOBAL_MODULES// } ]]; then
   echo "Reinstalling your global npm modules:"
   echo $GLOBAL_MODULES
-  npm install --global $GLOBAL_MODULES
+  if [[ $FORCE == 0 ]]; then
+    echo "Continue?"
+    select yn in "Yes" "No"; do
+      case $yn in
+        Yes ) npm install --global $GLOBAL_MODULES; break;;
+        No ) exit;;
+      esac
+    done
+  else
+    npm install --global $GLOBAL_MODULES
+  fi
+  
 fi
 
 if [[ $OSTYPE =~ "darwin" ]]; then

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "node-reinstall",
-  "version": "0.0.8",
+  "version": "0.0.9",
   "description": "Completely re-install Node, NPM and NVM",
   "scripts": [ "node-reinstall.sh" ],
   "install": "make install",

